File: php/src/Advertisements/User/Domain/Exceptions/UnauthorizedUserException.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\Domain\Exceptions;

use Demo\App\Common\Domain\DomainException;

final class UnauthorizedUserException extends DomainException
{
    private const string USER_UNAUTHORIZED = 'User unauthorized';

    private function __construct(string $message )
    {
        parent::__construct($message);
    }

    public static function build(): self
    {
        return new self(self::USER_UNAUTHORIZED);
    }

    public function message(): string
    {
        return $this->getMessage();
    }
}

File: kotlin/src/main/kotlin/advertisements/user/domain/exceptions/UnauthorizedUserException.kt
package advertisements.user.domain.exceptions

import common.domain.DomainException

class UnauthorizedUserException private constructor(message: String) : DomainException(message) {

    companion object {
        private const val USER_UNAUTHORIZED = "User unauthorized"

        fun build(): UnauthorizedUserException {
            return UnauthorizedUserException(USER_UNAUTHORIZED)
        }
    }

    fun message(): String {
        return this.message ?: ""
    }
}
